core/pkg/sync/kubernetes/kubernetes_sync.go

```diff
@@ -191,21 +192,29 @@ func (k *Sync) fetch(ctx context.Context) (string, error) {
 		}
 
 		k.logger.Debug(fmt.Sprintf("resource %s served from the informer cache", k.URI))
-		return configuration.Spec.FeatureFlagSpec, nil
+		b, err := json.Marshal(configuration.Spec.FlagSpec)
+		if err != nil {
+			return "", fmt.Errorf("failed to marshall FlagSpec: %s", err.Error())
+		}
+		return string(b), nil
```

Comment:
since we duplicate this, maybe worth considering moving them into its own function